### PR TITLE
[structopt] Add new port

### PR DESCRIPTION
**Pull request by p-ranav**
https://github.com/p-ranav/structopt

### CODE REVIEW

File: ports/structopt/CONTROL [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Source: structopt
+Version: v0.1.0
```
Comment (JackBoosY) [inline ports/structopt/CONTROL:2]:
```suggestion
Version: 0.1.0
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]
Does this port depends on other ports?

**Comment by p-ranav** [comment]
> Does this port depends on other ports?

`structopt` is self-contained although the source code comes packaged with a header each from the ports `magic-enum` and `visit-struct`.

**Comment by ras0219** [comment]
In order to avoid ODR violations, it's very important to have only one definition for any symbol across the entire vcpkg tree. In this case, this port should depend on `visit-struct` and `magic-enum` instead of vendoring its own copies.

**Comment by JackBoosY** [comment]
Pinging @p-ranav for response. Is work still being done for this PR?

**Comment by p-ranav** [comment]
At the moment, no. We can close this PR and I can create a new one when it's ready.

**Pull request closed without merging**